app/models/assignment_repo/creator.rb

```diff
@@ -189,5 +190,63 @@ def suffixed_repo_name(repository_name, suffix_count)
       suffix = "-#{suffix_count}"
       repository_name.truncate(100 - suffix.length, omission: '') + suffix
     end
+
+    # Handles assignment repository setup
+    #
+    # configs_repo    - GitHubRepository containing the configuration files
+    # assignment_repo - GitHubRepository for which to perform the configuration
+    #                   setups
+    #
+    # Returns nothing
+    def setup_repository(configs_repo, assignment_repo)
+      return unless configs_repo.classroom_configs?
+      conf_tree = configs_repo.branch('github-classroom').commit.commit.tree.sha
+      configs_repo.tree(conf_tree).tree.each do |conf|
+        process_config(configs_repo, assignment_repo, conf)
+      end
+
+      wait_import_completion assignment_repo
+
+      assignment_repo.remove_branch('github-classroom')
+    end
+
+    def wait_import_completion(github_repository)
```

Comment:
@tarebyte Do our heroku dynos only process one web request at a time?

If so, with the number of dynos we have, I'm not sure we can be freezing requests for up to 2 minutes. If a class releases an assignment to a roster of 300 people, it'll take a very small number (<10) of imports freezing to freeze the entire app for 2 minutes. On top of that, only a few students will be able to accept an invitation at once because of how slow the importer is.